src/frontend/src/lib/components/guard/AddressGuard.svelte

```diff
@@ -1,33 +1,43 @@
 <script lang="ts">
 	import { addressStore } from '$lib/stores/address.store';
-	import { isNullish } from '@dfinity/utils';
-	import { certifyEthAddress } from '$lib/services/address.services';
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { certifyBtcAddressMainnet, certifyEthAddress } from '$lib/services/address.services';
 	import { warnSignOut } from '$lib/services/auth.services';
-	import { ETHEREUM_TOKEN_ID } from '$env/tokens.env';
+	import {
+		addressesCertified,
+		btcAddressMainnet,
+		btcAddressMainnetNotCertified,
+		ethAddress,
+		ethAddressNotCertified
+	} from '$lib/derived/address.derived';
+	import { NETWORK_BITCOIN_ENABLED } from '$env/networks.btc.env.js';
 
 	const validateAddress = async () => {
-		const ethAddressData = $addressStore?.[ETHEREUM_TOKEN_ID];
-
-		if (isNullish(ethAddressData)) {
-			// No address is loaded, we don't have to verify it
-			return;
-		}
-
-		const { certified, data: ethAddress } = ethAddressData;
-
-		if (certified === true) {
-			// The address is certified, all good
+		if ($addressesCertified) {
+			// The addresses are certified, all good
 			return;
 		}
 
-		const { success, err } = await certifyEthAddress(ethAddress);
-
-		if (success) {
-			// The address is valid
-			return;
+		const results = await Promise.all([
+			NETWORK_BITCOIN_ENABLED && nonNullish($btcAddressMainnet) && $btcAddressMainnetNotCertified
+				? certifyBtcAddressMainnet($btcAddressMainnet)
+				: Promise.resolve({ success: true, err: null }),
+			nonNullish($ethAddress) && $ethAddressNotCertified
+				? certifyEthAddress($ethAddress)
+				: Promise.resolve({ success: true, err: null })
+		]);
+
+		let err: string | undefined = undefined;
+
+		results.map(({ success, err: e }) => {
```

Comment:
Same as in previous PR, can we use a non imperative approach?

Reply:
on it